# CSE 413 — Simulation and Modeling Lab
## Lab Assignment 2: Individualized Student Task — Unique Matrix Value Investigation

**Name:** Abdur Rahman Jayed  
**Student ID:** 0432320005101106  
**Course Code:** CSE 413

## Setup

In [1]:
import numpy as np

## TASK - 1: Create Your Personal Matrix

In [9]:
# TASK - 1
# Student ID: 0432320005101106
# Last two digits = 06
# d1 = last digit, d2 = second last digit

import numpy as np

d1 = 6
d2 = 0

A = np.array([[d1 + 2, d2 + 1],
              [2 * d1, d2 + 2]])

print("Matrix A:")
print(A)

Matrix A:
[[ 8  1]
 [12  2]]


## TASK - 2: Analyze Matrix A

In [10]:
# TASK - 2: Analyze Matrix A

print("Shape of A:", A.shape)

det_A = np.linalg.det(A)
print("Determinant of A:", det_A)

rank_A = np.linalg.matrix_rank(A)
print("Rank of A:", rank_A)

eigenvalues_A, eigenvectors_A = np.linalg.eig(A)
print("Eigenvalues of A:", eigenvalues_A)
print("Eigenvectors of A:\n", eigenvectors_A)

if det_A != 0:
    inverse_A = np.linalg.inv(A)
    print("Inverse of A:\n", inverse_A)
else:
    print("Inverse does not exist (determinant is 0)")

Shape of A: (2, 2)
Determinant of A: 3.999999999999999
Rank of A: 2
Eigenvalues of A: [9.58257569 0.41742431]
Eigenvectors of A:
 [[ 0.5341757  -0.13074917]
 [ 0.84537348  0.99141548]]
Inverse of A:
 [[ 0.5  -0.25]
 [-3.    2.  ]]


## TASK - 3: Value Change Experiment

In [12]:
# TASK - 3: Value Change Experiment
# I am changing element A[0,0] by adding +1
# Old value of A[0,0] = 8
# New value of B[0,0] = 9

B = A.copy()
B[0, 0] = B[0, 0] + 1

print("Changed element: A[0,0]")
print("Old value:", A[0, 0])
print("New value:", B[0, 0])
print("\nMatrix B:")
print(B)

Changed element: A[0,0]
Old value: 8
New value: 9

Matrix B:
[[ 9  1]
 [12  2]]


## TASK - 4: Re-analyze Matrix B

In [13]:
# TASK - 4: Analyze Matrix B

print("Shape of B:", B.shape)

det_B = np.linalg.det(B)
print("Determinant of B:", det_B)

rank_B = np.linalg.matrix_rank(B)
print("Rank of B:", rank_B)

eigenvalues_B, eigenvectors_B = np.linalg.eig(B)
print("Eigenvalues of B:", eigenvalues_B)
print("Eigenvectors of B:\n", eigenvectors_B)

if det_B != 0:
    inverse_B = np.linalg.inv(B)
    print("Inverse of B:\n", inverse_B)
else:
    print("Inverse does not exist (determinant is 0)")

Shape of B: (2, 2)
Determinant of B: 6.0
Rank of B: 2
Eigenvalues of B: [10.4244289  0.5755711]
Eigenvectors of B:
 [[ 0.57458003 -0.11787487]
 [ 0.8184484   0.99302846]]
Inverse of B:
 [[ 0.33333333 -0.16666667]
 [-2.          1.5       ]]


## TASK - 5: Comparison & Explanation

**1. How did the determinant change and why?**

Ans: The determinant increased from 4 to 6. This happened because I increased the value of A[0,0] from 8 to 9.

**2. Did the rank change? Explain.**

Ans: No, the rank did not change. The rank of both A and B is 2. Changing only one value did not affect the rank.


**3. How did the eigenvalues respond to the value change?**

Ans: The eigenvalues changed after changing one value. For A, they were about 9.58 and 0.42, while for B they became about 10.42 and 0.58.


**4. Is B easier or harder to invert than A? Why?**

Ans: B is slightly easier to invert because its determinant is 6, which is farther from zero than A's determinant of 4. Both matrices are invertible.